Move all TensorPipe code in single file, avoiding preprocessor flag

lw opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #42310 Don't include TensorPipe headers in public headers
* **#42299 Move all TensorPipe code in single file, avoiding preprocessor flag**

Differential Revision: [D22840087](https://our.internmc.facebook.com/intern/diff/D22840087/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D22840087/)!

lw committed:
Move all TensorPipe code in single file, avoiding preprocessor flag



Differential Revision: [D22840087](https://our.internmc.facebook.com/intern/diff/D22840087/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D22840087/)!

[ghstack-poisoned]

mrshenli reviewed (changes requested):
Oops, there are build failures

lw commented:
Sorry, this was just a PR used to trigger the CI while I was still iterating on the change. Let me mark it as a draft to avoid confusion.

lw committed:
Update on "Move all TensorPipe code in single file, avoiding preprocessor flag"


Differential Revision: [D22840087](https://our.internmc.facebook.com/intern/diff/D22840087/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D22840087/)!

[ghstack-poisoned]
